test/Microsoft.WebTools.AspireService.Tests/AspireServerServiceTests.cs

```diff
@@ -352,53 +351,37 @@ private async Task listenForSessionUpdatesAsync(AspireServerService aspireServer
                 {
                     await ws.CloseAsync(WebSocketCloseStatus.NormalClosure, null, CancellationToken.None);
                     return;
                 }
-                else
-                {
-                    var notificationBase = JsonSerializer.Deserialize<SessionNotificationBase>(message, AspireServerService.JsonSerializerOptions);
-                    if (notificationBase is null)
-                    {
-                        Console.WriteLine("Unexpected null SessionNotificationBase message");
-                    }
-                    else if (notificationBase.NotificationType == NotificationType.ProcessRestarted || notificationBase.NotificationType == NotificationType.SessionTerminated)
-                    {
-                        var scn = JsonSerializer.Deserialize<SessionChangeNotification>(message, AspireServerService.JsonSerializerOptions);
-                        if (scn is null)
-                        {
-                            Assert.Fail("Unexpected null SessionChangeNotification message");
-                        }
-                        else
-                        {
-                            callback.Invoke(scn);
-                        }
-                    }
-                    else if (notificationBase.NotificationType == NotificationType.ServiceLogs)
-                    {
-                        var sessionLogs = JsonSerializer.Deserialize<SessionLogsNotification>(message, AspireServerService.JsonSerializerOptions);
-                        if (sessionLogs is null)
-                        {
-                            Assert.Fail("Unexpected null SessionLogsNotification message");
-                        }
-                        else
-                        {
-                            callback.Invoke(sessionLogs);
-                        }
-                    }
-                }
             }
             catch
             {
                 // This is expected if the connection is closed
+                Assert.Equal(WebSocketState.Closed, ws.State);
                 return;
             }
+
+            var notification = JsonSerializer.Deserialize<SessionNotification>(message, AspireServerService.JsonSerializerOptions);
+            Assert.NotNull(notification);
+
+            SessionNotification value = notification.NotificationType switch
+            {
+                NotificationType.ProcessRestarted => JsonSerializer.Deserialize<ProcessRestartedNotification>(message, AspireServerService.JsonSerializerOptions),
+                NotificationType.SessionTerminated => JsonSerializer.Deserialize<SessionTerminatedNotification>(message, AspireServerService.JsonSerializerOptions),
+                NotificationType.ServiceLogs => JsonSerializer.Deserialize<ServiceLogsNotification>(message, AspireServerService.JsonSerializerOptions),
+                _ => throw new InvalidOperationException($"Unexpected {notification.NotificationType}")
```

Comment:
Maybe this shouldn't crash watch? It should instead log an error message but keep working?

Reply:
Oh wait this is a test 😄

Reply:
Crash is better here. This should never happen.